File: rpnumerics/physics/Quad3.java
/*
 * Instituto de Matematica Pura e Aplicada - IMPA
 * Departamento de Dinamica dos Fluidos
 *
 */

package rpnumerics.physics;

import wave.multid.Space;
import wave.multid.Multid;
import wave.util.Boundary;
import wave.util.RectBoundary;
import wave.util.RealVector;

public class Quad3 implements Physics
{


    //
    // Constants
    //
    static public final String FLUX_ID = "Quadratic R3";
    static public final RectBoundary DEFAULT_BOUNDARY = new RectBoundary(new RealVector(
        new double[] { -.5, -.5, -.5 }), new RealVector(
        new double[] { .5, .5, .5 }));

    //
    // Members
    //
    private FluxFunction fluxFunction_;

private AccumulationFunction accumulationFunction_;
    private Boundary boundary_=DEFAULT_BOUNDARY;

    //
    // Constructors/Inner Classes
    //
    public Quad3(Quad3FluxParams params) {

	fluxFunction_ = new QuadNDFluxFunction(params);
        accumulationFunction_= new DefaultAccFunction();
    }

    //
    // Accessors/Mutators
    //
    public Space domain() { return Multid.SPACE; }
    public FluxFunction fluxFunction( ){return fluxFunction_;}
    public String ID() { return FLUX_ID; }
//    public Boundary boundary() { return DEFAULT_BOUNDARY; }

    public void setBoundary (Boundary boundary) {boundary_=boundary;}

 public Boundary boundary() {return boundary_;}

 public  FluxFunction [] fluxFunctionArray(){return null;}

    public FluxFunction fluxFunction(int index){return null;}


    public AccumulationFunction accumulationFunction(){return accumulationFunction_;}



}

File: src/rpnumerics/physics/quad/Quad3.cc
#include "Quad3.h"

Quad3::Quad3(const Quad3FluxParams & params) : fluxFunction_(new Quad3FluxFunction(params)), accumulationFunction_(new Quad3AccumulationFunction()),
boundary_(defaultBoundary()) {
    FLUX_ID = "QuadraticR3";
    DEFAULT_SIGMA = "-.021";
    DEFAULT_XZERO = ".13 .07";
    printf("Hey\n");
}

const char * Quad3::ID(void) const {
    return FLUX_ID;
}

Quad3::~Quad3() {

    delete fluxFunction_;
    delete accumulationFunction_;
    delete boundary_;
}

Quad3::Quad3(const Quad3 & copy) {

    fluxFunction_ = new Quad3FluxFunction(( Quad3FluxFunction&)copy.fluxFunction());
    accumulationFunction_ = new Quad3AccumulationFunction((Quad3AccumulationFunction &) copy.accumulation());
    boundary_ = copy.boundary().clone();
    FLUX_ID = "QuadraticR3";

}
